Homework3/task2.py

```diff
@@ -9,4 +10,4 @@
 for i in range(int(len(list) / 2 + 0.5)):
```

Comment:
```suggestion
for i in range((len(list) + 1) // 2):
```
Так ведь намного лучше, не правда ли?